src/utils/mkarray.h

```diff
@@ -44,6 +49,26 @@ inline mxArray* make_mx_array(const std::vector<std::string> &v) {
 	return data;
 }
 
+template<>
+inline mxArray* make_mx_array(const std::vector<nix::Value> &v) {
+	if (v.empty()) {
+		return nullptr;
+	}
+
+	nix::DataType ntype = v[0].type();
+	mxArray *data;
+	if (ntype == nix::DataType::Nothing) {
+		data = nullptr; //fixme: something else?
+	} else {
+		data = mxCreateCellMatrix(1, v.size());
+		for (size_t i = 0; i < v.size(); i++) {
+			mxSetCell(data, i, make_mx_array(v[i]));
+		}
+	}
+
+	return data;
+}
+
```

Comment:
I would just keep all the `mxArray *` creation in one place, so one knows where to look. The old code called `make_mx_array` for the individual values in the array too.